tests/pytests/test_multithread.py

```diff
@@ -503,47 +506,64 @@ def check_threads(expected_num_threads_alive, expected_n_threads):
 
     # Increase number of threads
     env.expect(config_cmd(), 'SET', 'WORKERS', '2').ok()
+    # After the first increase, since no queries arrived yet
     check_threads(expected_num_threads_alive=0, expected_n_threads=2)
     # Decrease number of threads
     env.expect(config_cmd(), 'SET', 'WORKERS', '1').ok()
     check_threads(expected_num_threads_alive=0, expected_n_threads=1)
+    # If I send many queries, we know one of the threads will take the ADMIN job and terminate
+    num_query_threads = 100
+    query_threads = []
+
+    for i in range(num_query_threads):
+        t = threading.Thread(target=send_query, name=f'QueryThread-{i}')
+        t.start()
+        query_threads.append(t)
+
+    for t in query_threads:
+        t.join()
+
+    check_threads(expected_num_threads_alive=1, expected_n_threads=1)
     # Set it to 0
     env.expect(config_cmd(), 'SET', 'WORKERS', '0').ok()
+    time.sleep(1)
     check_threads(expected_num_threads_alive=0, expected_n_threads=0)
 
     # Query should not be executed by the threadpool
-    env.expect('FT.CREATE', 'idx', 'SCHEMA', 't', 'text').ok()
+
     env.expect('ft.search', 'idx', '*').equal([0])
     check_threads(expected_num_threads_alive=0, expected_n_threads=0)
-    env.assertEqual(getWorkersThpoolStats(env)['totalJobsDone'], 0)
+    env.assertEqual(getWorkersThpoolStats(env)['totalJobsDone'], num_query_threads)
```

Comment:
Why has this changed? We previously wanted admin jobs not to affect this number

Reply:
because we have added query threads. Notice the jobs are based on query threads, not on ADMIN jobs